src/app-layer-parser.c

```diff
@@ -1438,6 +1441,32 @@ int AppLayerParserParse(ThreadVars *tv, AppLayerParserThreadCtx *alp_tctx, Flow
         }
     }
 
+    if (pstate->flags & APP_LAYER_PARSER_LAYERED_PACKET) {
+        // hardcoded logic
```

Comment:
TODO: instead of hardcoded logic, protocols (such as HTTP2) should have a generic function that returns the pseudo-packets payloads